submissions/api.py

```diff
@@ -764,6 +770,14 @@ def set_score(submission_uuid, points_earned, points_possible):
     try:
         score_model = score.save()
         _log_score(score_model)
+        if annotation_creator is not None:
+            score_annotation = ScoreAnnotation(
+                score=score_model,
+                creator=annotation_creator,
+                annotation_type=annotation_type,
+                reason=annotation_reason
+            )
+            score_annotation.save()
```

Comment:
We can end up with a score without an annotation if the annotation fails. Do you think it makes sense to make it a manual transaction where both have to succeed before commiting?